app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt

```diff
@@ -82,7 +90,6 @@ fun ReaderScreen(info: BaseGalleryInfo, page: Int = -1) {
     Deferred({ pageLoader.awaitReady() }) {
         val lazyListState = rememberLazyListState()
         val pagerState = rememberPagerState { pageLoader.size }
-        val zoomableState = rememberZoomableState(zoomSpec = ZoomSpec(maxZoomFactor = 3f))
```

Comment:
We may want it here for https://saket.github.io/telephoto/zoomable/#edge-detection

Reply:
Done

Reply:
Pager should only be zoomable when image is ready.

Reply:
In WEBTOON, CONTINUOUS_VERTICAL mode, this is only one zoomState and globally zoomable
In DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL we should have multiple zoomState but currently, one
Hard to implement this

Reply:
Pass page state to GalleryPager?

Reply:
I mean, we shouldn't do this now,
As we currently use a global zoomState so it is keeped during page swipe,
User must be able to reset the scale state regardless of page state

Reply:
That's another thing.
I was talking about the zoomable modifier should be applied on the image, progress indicator and error message shouldn't be zoomable.

Reply:
Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable,
But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image

Reply:
Let's implement it after https://github.com/FooIbar/EhViewer/issues/609

Reply:
> Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable, But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image

Fair.

Reply:
> > Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable, But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image
> 
> Fair.

Is https://github.com/revonateB0T/EhViewer/commit/eb7b8d5d29e7b7f25da9433fae0bdd8f8cf3fcb8 the one you mean?

Reply:
> > > Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable, But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image
> > 
> > 
> > Fair.
> 
> Is [revonateB0T@eb7b8d5](https://github.com/revonateB0T/EhViewer/commit/eb7b8d5d29e7b7f25da9433fae0bdd8f8cf3fcb8) the one you mean?

That looks overcomplicated.
What about moving `item` to a separate Composable that takes something like `isWebtoon`?

Reply:
> > > > Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable, But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image
> > > 
> > > 
> > > Fair.
> > 
> > 
> > Is [revonateB0T@eb7b8d5](https://github.com/revonateB0T/EhViewer/commit/eb7b8d5d29e7b7f25da9433fae0bdd8f8cf3fcb8) the one you mean?
> 
> That looks overcomplicated. What about moving `item` to a separate Composable that takes something like `isWebtoon`?

Then we have to make `zoomableState` controlled in different `Composable`

Reply:
> > > > > Yeah, I agree that in DEFAULT, LEFT_TO_RIGHT, RIGHT_TO_LEFT, VERTICAL, they shouldn't be zoomable, But in WEBTOON, CONTINUOUS_VERTICAL, they should have the scale per image
> > > > 
> > > > 
> > > > Fair.
> > > 
> > > 
> > > Is [revonateB0T@eb7b8d5](https://github.com/revonateB0T/EhViewer/commit/eb7b8d5d29e7b7f25da9433fae0bdd8f8cf3fcb8) the one you mean?
> > 
> > 
> > That looks overcomplicated. What about moving `item` to a separate Composable that takes something like `isWebtoon`?
> 
> Then we have to make `zoomableState` controlled in different `Composable`

The item Composable can have a default zoomableState parameter.
And we pass the parent‘s zoomableState in webtoon scenario.

Reply:
fa05bd75c2dad787ceefd94a543546e863a90b3f
Then we have to pass `onMenuRegionClick`, `onSelectPage` to PagerItem
And conditionally wrap with a `Box` or make `PagerItem` a local `@Composable`
I don't think it is much cleaner